### PR TITLE
Calibration

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/intake/neoIntake/NeoIntakeConstants.java [hunk 1]
```diff
@@ -7,11 +7,7 @@
 
 public class NeoIntakeConstants {
 	
-	public static final int INTAKE_ID = 62;
-	
-	public static final int ENTRANCE_BEAM_BREAKER_CHANNEL = 0;
-	
-	public static final int EXIT_BEAM_BREAKER_CHANNEL = 1;
+	public static final int INTAKE_ID = 3;
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/intake/neoIntake/NeoIntakeConstants.java:10]: rename to motor id

File: src/main/java/edu/greenblitz/robotName/commands/shooter/pivot/MovePivotToAngle.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import edu.greenblitz.robotName.Robot;
 import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/pivot/MovePivotToAngle.java:5]: unused import

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java [hunk 3]
```diff
@@ -74,12 +82,21 @@ public void moveToAngle(Rotation2d targetAngle) {
 				positionVoltage
 						.withPosition(targetAngle.getRotations())
 						.withSlot(FalconPivotConstants.PID_SLOT)
-						.withLimitForwardMotion(true)
-						.withLimitReverseMotion(true)
+						.withLimitForwardMotion(false)
+						.withLimitReverseMotion(false)
```
Comment (NoamRosenberg08) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java:86]: shouldnt it be true

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/falconFlyWheel/FalconFlyWheelConstants.java [hunk 4]
```diff
@@ -7,58 +7,46 @@ public class FalconFlyWheelConstants {
 	
 	public static class rightMotorConstants {
 		
-		public static final int ID = 12;
+		public static final int ID = 16;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/falconFlyWheel/FalconFlyWheelConstants.java:10]: change name to MOTOR_ID to follow convention

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/falconFlyWheel/FalconFlyWheelConstants.java [hunk 5]
```diff
@@ -7,58 +7,46 @@ public class FalconFlyWheelConstants {
...
 
-		public static final String CANBUS_CHAIN = "*";
+		public static final String CANBUS_CHAIN = "";
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/falconFlyWheel/FalconFlyWheelConstants.java:12]: not connected to canivore?

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnel.java [hunk 6]
```diff
@@ -34,6 +33,6 @@ public void setPower(double power) {
 	public void updateInputs(FunnelInputsAutoLogged inputs) {
 		inputs.outputCurrent = motor.getOutputCurrent();
 		inputs.appliedOutput = motor.getAppliedOutput();
-		inputs.isObjectIn = debouncer.calculate(beamBreaker.get());
+		inputs.isObjectIn = debouncer.calculate(motor.getReverseLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyOpen).isLimitSwitchEnabled());
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnel.java:36]: test to see if this also updates the `this.isObjectIn`

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java [hunk 7]
```diff
@@ -24,12 +30,14 @@ public FalconPivot() {
...
 		motor.setNeutralMode(FalconPivotConstants.NEUTRAL_MODE_VALUE);
+		motor.setInverted(FalconPivotConstants.INVERTED);
 		optimizeCanBusUtilization();
 		
 		absoluteEncoder = new DutyCycleEncoder(FalconPivotConstants.ABSOLUTE_ENCODER_CHANNEL);
-		
-		resetAngle(Rotation2d.fromRotations(absoluteEncoder.getAbsolutePosition()));
-		
+
+		getAbsoluteEncoderPosition();
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java:38]: delete

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java [hunk 8]
```diff
@@ -24,12 +30,14 @@ public FalconPivot() {
...
+		resetAngle(getAbsoluteEncoderPosition());
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java:39]: not here

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java [hunk 9]
```diff
@@ -88,9 +105,12 @@ public void updateInputs(PivotInputsAutoLogged inputs) {
 		inputs.position = Rotation2d.fromRotations(motor.getPosition().getValue());
 		inputs.velocity = motor.getVelocity().getValue();
 		inputs.acceleration = motor.getAcceleration().getValue();
-		inputs.absoluteEncoderPosition = absoluteEncoder.getAbsolutePosition();
+		inputs.absoluteEncoderPosition = getAbsoluteEncoderPosition();
 		inputs.temperature = motor.getDeviceTemp().getValue();
 		inputs.hasHitForwardLimit = motor.getForwardLimit().getValue().value == IS_SWITCH_CLOSED;
 		inputs.hasHitBackwardsLimit = motor.getReverseLimit().getValue().value == IS_SWITCH_CLOSED;
+
+		SmartDashboard.putNumber("abs pos", inputs.absoluteEncoderPosition.getDegrees());
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java:113]: remove

**Review by yonnbro — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnel.java (hunk 6 above)
Comment (yonnbro) [reply src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnel.java:36]: ok

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/falconFlyWheel/FalconFlyWheelConstants.java (hunk 5 above)
Comment (yonnbro) [reply src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/falconFlyWheel/FalconFlyWheelConstants.java:12]: mo

**Review by yonnbro — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnel.java [hunk 10]
```diff
@@ -34,6 +34,7 @@ public void setPower(double power) {
 	public void updateInputs(FunnelInputsAutoLogged inputs) {
 		inputs.outputCurrent = motor.getOutputCurrent();
 		inputs.appliedOutput = motor.getAppliedOutput();
-		inputs.isObjectIn = debouncer.calculate(beamBreaker.get());
+		inputs.isObjectIn = debouncer.calculate(motor.getReverseLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyOpen).isPressed());
+		SmartDashboard.putBoolean("funnel",inputs.isObjectIn);
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/funnel/neoFunnel/NeoFunnel.java:38]: No no no no

**Review by yonnbro — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/intake/neoIntake/NeoIntake.java [hunk 11]
```diff
@@ -45,8 +47,13 @@ public void updateInputs(IntakeInputsAutoLogged intakeInputs) {
 		intakeInputs.entranceBeamBreakerValue = entranceBeamBreaker.calculate(
 				motor.getReverseLimitSwitch(NeoIntakeConstants.ENTRANCE_BEAM_BREAKER_TYPE).isPressed()
 		);
-		intakeInputs.exitBeamBreakerValue = exitBeamBreaker.calculate(
-				motor.getForwardLimitSwitch(NeoIntakeConstants.EXIT_BEAM_BREAKER_TYPE).isPressed()
-		);
+//		intakeInputs.exitBeamBreakerValue = exitBeamBreaker.calculate(
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/intake/neoIntake/NeoIntake.java:50]: a lot of comments here, delete if really or used or uncomment

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 12]
```diff
@@ -137,7 +137,7 @@ private void initializeLogger() {
 
 	@Override
 	public void autonomousInit() {
-		AutonomousSelector.getInstance().getChosenValue().schedule();
+//		AutonomousSelector.getInstance().getChosenValue().schedule();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/Robot.java:140]: uncomment

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by yonnbro — APPROVED** [review]

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 13]
```diff
@@ -47,8 +45,31 @@ private OI() {
...
+		mainJoystick.POV_UP.whileTrue(
+				new RunCommand(() -> frontLeft.rotateToAngle(Rotation2d.fromDegrees(360))));
+		mainJoystick.POV_RIGHT.whileTrue(
+				new RunCommand(() -> frontLeft.rotateToAngle(Rotation2d.fromDegrees(270))));
+		mainJoystick.POV_LEFT.whileTrue(
+				new RunCommand(() -> frontLeft.rotateToAngle(Rotation2d.fromDegrees(180))));
+		mainJoystick.POV_DOWN.whileTrue(
+				new RunCommand(() -> frontLeft.rotateToAngle(Rotation2d.fromDegrees(90))));
+		mainJoystick.R1.whileTrue(
+				new RunCommand(() -> frontLeft.rotateToAngle(Rotation2d.fromDegrees(90))));
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:69]: huh?

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 14]
```diff
@@ -47,8 +45,31 @@ private OI() {
...
+
+//		initButtons();
+//		initializeDefaultCommands();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:72]: dont mark it resolved if its not..

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 15]
```diff
@@ -47,8 +45,31 @@ private OI() {
 		thirdJoystick = new SmartJoystick(RobotConstants.Joystick.THIRD);
 		fourthJoystick = new SmartJoystick(RobotConstants.Joystick.FOURTH);
 
-		initButtons();
-		initializeDefaultCommands();
+		secondJoystick.B.whileTrue(new RunFunnelByPowerUntilCondition(
+				0.3,
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:49]: magic..

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 16]
```diff
@@ -79,30 +100,33 @@ public SmartJoystick getFourthJoystick() {
...
-		mainJoystick.A.whileTrue(new GoToShootingState(ShootingPositionConstants.OPTIMAL_SHOOTING_ZONE));
+//		secondJoystick.START.whileTrue(new PanicMode());
+//		secondJoystick.BACK.whileTrue(new ToggleScoringMode());
+//		secondJoystick.A.whileTrue(new ScoreToAmp());
+//		secondJoystick.B.whileTrue(new MoveElbowAndWristToSafe());
+//		secondJoystick.X.whileTrue(new MoveShooterToAngle(PivotConstants.PresetPositions.PICK_UP.ANGLE));
+//		secondJoystick.Y.whileTrue(new MoveShooterToAngle(PivotConstants.PresetPositions.TRANSFER.ANGLE));
+//		secondJoystick.POV_DOWN.whileTrue(new CollectNoteToScoringMode());
+//
+//		mainJoystick.A.whileTrue(new GoToShootingState(ShootingPositionConstants.OPTIMAL_SHOOTING_ZONE));
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:111]: give back...

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 17]
```diff
@@ -79,30 +100,33 @@ public SmartJoystick getFourthJoystick() {
...
 	public void initializeDefaultCommands() {
 		SwerveChassis.getInstance().setDefaultCommand(new MoveByJoysticks(ChassisConstants.DRIVE_MODE));
-		Battery.getInstance().setDefaultCommand(new BatteryLimiter());
-		Elbow.getInstance().setDefaultCommand(new ElbowDefaultCommand());
-		Wrist.getInstance().setDefaultCommand(new WristDefaultCommand());
-		Pivot.getInstance().setDefaultCommand(new PivotDefaultCommand());
+//		Battery.getInstance().setDefaultCommand(new BatteryLimiter());
+//		Elbow.getInstance().setDefaultCommand(new ElbowDefaultCommand());
+//		Wrist.getInstance().setDefaultCommand(new WristDefaultCommand());
+//		Pivot.getInstance().setDefaultCommand(new PivotDefaultCommand());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:130]: stop resolving and not solving..

File: src/main/java/edu/greenblitz/robotName/Robot.java
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/Robot.java:1]: so much commented out code

File: src/main/java/edu/greenblitz/robotName/commands/NoteToShooter.java [hunk 18]
```diff
@@ -30,6 +42,7 @@ public boolean isFinished() {
 
     @Override
     public void end(boolean interrupted) {
+        SmartDashboard.putNumber("SS", 1);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/NoteToShooter.java:45]: delete

File: src/main/java/edu/greenblitz/robotName/commands/NoteToShooter.java [hunk 19]
```diff
@@ -2,24 +2,36 @@
...
+    
+    @Override
+    public void initialize() {
+        timer.restart();
+    }
+    
     @Override
     public void execute() {
-        funnel.rollIn();
+        if (timer.get() > FunnelConstants.NOTE_TO_SHOOTER_DELAY_TIME)
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/NoteToShooter.java:33]: add {

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java [hunk 20]
```diff
@@ -80,17 +85,28 @@ public void moveToAngle(Rotation2d targetAngle) {
...
+		inputs.velocity = motor.getLatencyCompensatedValue(motor.getPosition(), motor.getVelocity());
 		inputs.acceleration = motor.getAcceleration().getValue();
-		inputs.absoluteEncoderPosition = absoluteEncoder.getAbsolutePosition();
+		inputs.absoluteEncoderPosition = getAbsoluteEncoderPosition();
 		inputs.temperature = motor.getDeviceTemp().getValue();
 		inputs.hasHitForwardLimit = motor.getForwardLimit().getValue().value == IS_SWITCH_CLOSED;
 		inputs.hasHitBackwardsLimit = motor.getReverseLimit().getValue().value == IS_SWITCH_CLOSED;
+
+		SmartDashboard.putNumber("pivit angle", inputs.position.getDegrees());
+		SmartDashboard.putNumber("pivit angle abs", inputs.absoluteEncoderPosition.getDegrees());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivot.java:110]: dym abs like absurd or abstract?

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java [hunk 21]
```diff
@@ -34,11 +43,13 @@ public class FalconPivotConstants {
...
 		SLOT_0_CONFIGS.kV = kV;
-		SLOT_0_CONFIGS.kP = 0.129;
+		SLOT_0_CONFIGS.kP = 198;
 		SLOT_0_CONFIGS.kI = 0;
-		SLOT_0_CONFIGS.kD = 0;
+		SLOT_0_CONFIGS.kD = 3.6;
 	}
-	
+
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java:51]: delete

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java [hunk 22]
```diff
@@ -58,17 +69,22 @@ public class FalconPivotConstants {
 	}
 	
 	public static final FeedbackConfigs FEEDBACK_CONFIGS = new FeedbackConfigs();
-	
 	static {
 		FEEDBACK_CONFIGS.SensorToMechanismRatio = PivotConstants.GEAR_RATIO;
 	}
+
+	public static final HardwareLimitSwitchConfigs HARDWARE_LIMIT_SWITCH_CONFIGS = new HardwareLimitSwitchConfigs();
+	static {
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java:77]: add a space

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java [hunk 23]
```diff
@@ -58,17 +69,22 @@ public class FalconPivotConstants {
...
+		HARDWARE_LIMIT_SWITCH_CONFIGS.ReverseLimitEnable = true;
+		HARDWARE_LIMIT_SWITCH_CONFIGS.ReverseLimitType = ReverseLimitTypeValue.NormallyOpen;
+	}
 	
 	public static final TalonFXConfiguration TALON_FX_CONFIGURATION = new TalonFXConfiguration();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/pivot/falconPivot/FalconPivotConstants.java:82]: add a space

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java [hunk 24]
```diff
@@ -40,42 +40,41 @@ public class MK4iSwerveConstants {
 	public static final TalonFXConfiguration ANGULAR_FALCON_CONFIG_OBJECT = new TalonFXConfiguration();
 	
 	static {
+		FeedbackConfigs FEEDBACK_CONFIGS = new FeedbackConfigs();
+		FEEDBACK_CONFIGS.SensorToMechanismRatio = ANGULAR_GEAR_RATIO;
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:45]: delete space

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java [hunk 25]
```diff
@@ -40,42 +40,41 @@ public class MK4iSwerveConstants {
...
+
 		Slot0Configs PID_CONFIGS = new Slot0Configs();
-		PID_CONFIGS.kP = 1;
+		PID_CONFIGS.kP = 0;//70;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:48]: //70?

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java [hunk 26]
```diff
@@ -40,42 +40,41 @@ public class MK4iSwerveConstants {
...
 		PID_CONFIGS.kI = 0;
 		PID_CONFIGS.kD = 0;
-		PID_CONFIGS.kS = 0;
+		PID_CONFIGS.kS = ks;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:51]: why is only the ks = ks?

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+    protected static final double DYNAMIC_VOLTAGE = 1;
+
+    protected static final int MOTOR_ID = 5;
+
+    protected static final String CANBUS_CHAIN = RobotConstants.General.CANIVORE_NAME;
+
+    public static final SoftwareLimitSwitchConfigs SOFTWARE_LIMIT_SWITCH_CONFIGS = new SoftwareLimitSwitchConfigs();
+    static {
+
+    }
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java:23]: SROP RSOLVING AND NOT SOLVING, and why is it empty

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java [hunk 28]
```diff
@@ -0,0 +1,35 @@
...
+
+    public static final HardwareLimitSwitchConfigs HARDWARE_LIMIT_SWITCH_CONFIGS = new HardwareLimitSwitchConfigs();
+    static {
+
+    }
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java:28]: same here

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java [hunk 29]
```diff
@@ -0,0 +1,35 @@
...
+
+    public static final FeedbackConfigs FEEDBACK_CONFIGS = new FeedbackConfigs();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java:30]: add a space means to the line right under my comment

File: src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java [hunk 30]
```diff
@@ -0,0 +1,35 @@
...
+    static {
+        FEEDBACK_CONFIGS.SensorToMechanismRatio = MK4iSwerveConstants.ANGULAR_GEAR_RATIO;
+    }
+
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/utils/SysId/falconSysId/SysIdFalconConstants.java:34]: delete space

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by yonnbro — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Pull request merged**